app/models/organization/CreditTransactionService.scala

```diff
@@ -68,6 +69,12 @@ class CreditTransactionService @Inject()(creditTransactionDAO: CreditTransaction
       implicit ctx: DBAccessContext): Fox[Unit] =
     creditTransactionDAO.addJobIdToTransaction(creditTransaction, jobId)
 
+  def findTransactionOfJob(jobId: ObjectId)(implicit ctx: DBAccessContext): Fox[Option[CreditTransaction]] =
+    creditTransactionDAO.findTransactionForJob(jobId).futureBox.flatMap {
+      case Full(transaction) => Fox.successful(Some(transaction))
+      case _                 => Fox.successful(None)
+    }
```

Comment:
if you replace flatMap by map, you can leave out the Fox.successful.
I feel like we have a box2Option conversion somewhere that does pretty much this. Maybe you could go on a search, or implement it if it is missing. Maybe box even has a toOption?